src/ngx_http_lua_api.c

```diff
@@ -213,4 +216,74 @@ ngx_http_lua_shared_memory_init(ngx_shm_zone_t *shm_zone, void *data)
     return NGX_OK;
 }
 
+
+ngx_int_t
+ngx_http_lua_delay_list_register(ngx_conf_t *cf, void *tag)
+{
+    ngx_http_lua_main_conf_t     *lmcf;
+    void                        **init;
+    unsigned                      i;
+
+    lmcf = ngx_http_conf_get_module_main_conf(cf, ngx_http_lua_module);
+    if (lmcf == NULL) {
+        return NGX_ERROR;
+    }
+
+    if (lmcf->delay_list == NULL) {
+        lmcf->delay_list =
+            ngx_array_create(cf->pool, 4, sizeof(void *));
+
+        if (lmcf->delay_list == NULL) {
+            return NGX_ERROR;
+        }
+    }
+
+    init = lmcf->delay_list->elts;
+
+    for (i = 0; i < lmcf->delay_list->nelts; i++) {
+
+        if (init[i] == tag) {
+            return NGX_OK;
+        }
+    }
+
+    init = ngx_array_push(lmcf->delay_list);
+    if (init == NULL) {
+        return NGX_ERROR;
+    }
+
+    *init = tag;
+
+    return NGX_OK;
+}
+
+
+ngx_int_t
+ngx_http_lua_delay_init_phase()
+{
+    ngx_http_lua_main_conf_t     *lmcf;
+
+    lmcf = ngx_http_cycle_get_module_main_conf(ngx_cycle, ngx_http_lua_module);
+    if (lmcf == NULL) {
+        return NGX_ERROR;
+    }
+
+    if (!lmcf->init_handler) {
+        return NGX_OK;
+    }
+
+    lmcf->delay_list_inited++;
+
+    if (lmcf->delay_list->nelts > lmcf->delay_list_inited) {
```

Comment:
@agentzh Because `ngx_http_lua_delay_init_phase` will  run in third part module, I think that if there are third party modules called `ngx_http_lua_delay_init_phase` too much times, that is, the number of times the `init` more than the number of `register`, the error should be returned.